code/modules/mob/living/carbon/xenomorph/castes/Crusher.dm

```diff
@@ -79,100 +79,84 @@
+		var/obj/structure/machinery/door/airlock/airlock_in_path = target
 
-		if (A.unacidable)
+		if (airlock_in_path.unacidable)
 			. = FALSE
 		else
-			A.deconstruct()
+			airlock_in_path.deconstruct()
 
 	else if (istype(target, /obj/structure/grille))
-		var/obj/structure/grille/G = target
-		if(G.unacidable)
+		var/obj/structure/grille/grille_in_path = target
+		if(grille_in_path.unacidable)
 			. =  FALSE
 		else
-			G.health -=  80 //Usually knocks it down.
-			G.healthcheck()
+			grille_in_path.health -=  80 //Usually knocks it down.
+			grille_in_path.healthcheck()
 			. = TRUE
 
 	else if (istype(target, /obj/structure/surface/table))
-		var/obj/structure/surface/table/T = target
-		T.Crossed(src)
+		var/obj/structure/surface/table/table_in_path = target
+		table_in_path.Crossed(src)
 		. = TRUE
 
 	else if (istype(target, /obj/structure/machinery/defenses))
-		var/obj/structure/machinery/defenses/DF = target
-		visible_message(SPAN_DANGER("[src] rams [DF]!"), SPAN_XENODANGER("We ram [DF]!"))
+		var/obj/structure/machinery/defenses/defenses_in_path = target
+		visible_message(SPAN_DANGER("[src] rams [defenses_in_path]!"), SPAN_XENODANGER("We ram [defenses_in_path]!"))
 
-		if (!DF.unacidable)
+		if (!defenses_in_path.unacidable)
 			playsound(loc, "punch", 25, 1)
-			DF.stat = 1
-			DF.update_icon()
-			DF.update_health(40)
+			defenses_in_path.stat = 1
+			defenses_in_path.update_icon()
+			defenses_in_path.update_health(40)
 
 		. =  FALSE
 
-	else if (istype(target, /obj/structure/machinery/vending))
-		var/obj/structure/machinery/vending/V = target
+	else if (istype(target, /obj/structure/machinery/vending && /obj/structure/machinery/cm_vending))
+		var/obj/structure/machinery/vending/vending_in_path = target
 
-		if (V.unslashable)
+		if (vending_in_path.unslashable)
 			. = FALSE
 		else
-			visible_message(SPAN_DANGER("[src] smashes straight into [V]!"), SPAN_XENODANGER("We smash straight into [V]!"))
+			visible_message(SPAN_DANGER("[src] smashes straight into [vending_in_path]!"), SPAN_XENODANGER("We smash straight into [vending_in_path]!"))
 			playsound(loc, "punch", 25, 1)
-			V.tip_over()
+			vending_in_path.tip_over()
 
 			var/impact_range = 1
-			var/turf/TA = get_diagonal_step(V, dir)
-			TA = get_step_away(TA, src)
+			var/turf/vending_turf = get_diagonal_step(vending_in_path, dir)
+			vending_turf = get_step_away(vending_turf, src)
 			var/launch_speed = 2
-			launch_towards(TA, impact_range, launch_speed)
-
-			. =  TRUE
-
-	else if (istype(target, /obj/structure/machinery/cm_vending))
-		var/obj/structure/machinery/cm_vending/V = target
-		if (V.unslashable)
-			. = FALSE
-		else
-			visible_message(SPAN_DANGER("[src] smashes straight into [V]!"), SPAN_XENODANGER("We smash straight into [V]!"))
-			playsound(loc, "punch", 25, 1)
-			V.tip_over()
-
-			var/impact_range = 1
-			var/turf/TA = get_diagonal_step(V, dir)
-			TA = get_step_away(TA, src)
-			var/launch_speed = 2
-			throw_atom(TA, impact_range, launch_speed)
```

Comment:
where it go

Reply:
i merged them into one at 145 if you had actually read the code buddy pal